### PR TITLE
Update ics to indicate support for OOB pairing and add support for SM/MAS/SCOB/BV-01-C

### PR DESCRIPTION
**Pull request by kromancer**
Also check:
https://github.com/AyturkDuzen/zephyr/pull/2

In the zip you can also find an air trace:
[SM_MAS_SCOB_BV_01_C_2020_10_13_13_05_45.zip](https://github.com/AyturkDuzen/auto-pts/files/5370941/SM_MAS_SCOB_BV_01_C_2020_10_13_13_05_45.zip)

### CODE REVIEW

File: ptsprojects/zephyr/sm_wid.py [hunk 1]
```diff
@@ -112,6 +113,51 @@ def hdl_wid_143(desc):
...
+    """
+    Please configure IUT's OOB flag with 'Remote OOB data present'
+
+    TODO: The flag will be set when we handle wid 149 - set remote oob data
+    """
+    return True
+
+
+def hdl_wid_147(desc):
+    """
```
Comment (AyturkDuzen) [inline ptsprojects/zephyr/sm_wid.py:136]: {147,%s,SM}Please enter 16 bytes IUT's OOB Data (confirmation).
{148,%s,SM}Please enter 16 bytes IUT's OOB Key (random number).

Assumption 1) R and C can not be used alone and they are always required together. 
Assumption 2) OOB sc local data is fixed.
With assumption 1 and 2, should we call this function two times? We can save execution time by calling it once and set r and c locally. If you want to keep it this way I can understand for integrity of the wid implementation.

File: ptsprojects/zephyr/sm.py [hunk 2]
```diff
@@ -335,6 +335,11 @@ def test_cases(pts):
                   pre_conditions +
                   [TestFunc(btp.gap_set_io_cap, IOCap.no_input_output),],
                   generic_wid_hdl=sm_wid_hdl),
+
```
Comment (AyturkDuzen) [inline ptsprojects/zephyr/sm.py]: remove the empty line for integrity also on line 333.

Comment (kromancer) [reply ptsprojects/zephyr/sm.py]: will do

File: ptsprojects/zephyr/sm_wid.py (hunk 1 above)
Comment (kromancer) [reply ptsprojects/zephyr/sm_wid.py:136]: I agree, it's a bit confusing why they have 2 different wids for the OOB.
I just copied over the implementation from mynewt, maybe we later find out that there is a reason for it.

For the time being, how about just adding a TODO comment?

File: ptsprojects/zephyr/sm.py [hunk 3]
```diff
@@ -330,11 +330,14 @@ def test_cases(pts):
                   pre_conditions +
                   [TestFunc(btp.gap_set_io_cap, IOCap.keyboard_display)],
                   generic_wid_hdl=sm_wid_hdl),
-
         ZTestCase("SM", "SM/SLA/SIE/BV-01-C",
                   pre_conditions +
                   [TestFunc(btp.gap_set_io_cap, IOCap.no_input_output),],
                   generic_wid_hdl=sm_wid_hdl),
+        ZTestCase("SM", "SM/MAS/SCOB/BV-01-C",
+                  pre_conditions +
```
Comment (AyturkDuzen) [inline ptsprojects/zephyr/sm.py:340]: Is there any prj.config that we need to enable for this test case?

If so should we add like this:
        ZTestCase("SM", "SM/MAS/OOB/BV-01-C",
                  # Set CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY=y in prj.conf
                  pre_conditions +

Comment (kromancer) [reply ptsprojects/zephyr/sm.py:340]: We are lucky - no special `CONFIG_` is required :)

File: ptsprojects/zephyr/sm_wid.py (hunk 1 above)
Comment (kromancer) [reply ptsprojects/zephyr/sm_wid.py:136]: or not :)

**Review by ryanjh — COMMENTED** [review]
You can think of using # for the comments as same as wid_116.

File: ptsprojects/zephyr/sm_wid.py (hunk 1 above)
Comment (AyturkDuzen) [reply ptsprojects/zephyr/sm_wid.py:136]: for now lets leave it like this

**Review by AyturkDuzen — APPROVED** [review]

**Review by ryanjh — APPROVED** [review]

File: ptsprojects/zephyr/sm.py [hunk 4]
```diff
@@ -239,7 +239,6 @@ def test_cases(pts):
                   [TestFunc(btp.gap_set_io_cap, IOCap.display_only)],
                   generic_wid_hdl=sm_wid_hdl),
         ZTestCase("SM", "SM/MAS/OOB/BV-01-C",
-                  # Set CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY=y in prj.conf
```
Comment (kromancer) [inline ptsprojects/zephyr/sm.py:242]: @AyturkDuzen , check with:
https://github.com/AyturkDuzen/zephyr/pull/2

It seems that we do not need the kconfig anymore

Comment (AyturkDuzen) [reply ptsprojects/zephyr/sm.py:242]: yes I also tried with both softdevice controller and software based. No need for extra config

**Review by AyturkDuzen — APPROVED** [review]

**Pull request merged**